### ISSUE TITLE
"list index out of range" error when validation set is larger than test set in TextClasDataBunch.create

### ISSUE THREAD

**Issue by ohmeow** [fastai/fastai#1299]
<img width="1017" alt="screen shot 2018-12-06 at 12 41 17 am" src="https://user-images.githubusercontent.com/14000/49572025-b574c100-f8ef-11e8-8cb6-325b721823f8.png">

If you include a test dataset and it is smaller than the validation dataset, you will get an "list index out of range" exception when trying to iterate over the validation dataloader.

I don't know enough about python here but it looks like something is being passed around by reference in the create method of the TextClasDataBunch class.  In particular this section:

```
for ds in datasets[1:]:
            lengths = [len(t) for t in ds.x.items]
            sampler = SortSampler(ds.x, key=lambda t: lengths[t])
            dataloaders.append(DataLoader(ds, batch_size=bs, sampler=sampler, **kwargs))
```
When you attempt to iterate over the validation dataloader, it use the number of items in the test set to construct it's `SortSampler` for some reason, thus throwing the "index out of range" exception when there are more items in the validation set than in the test set.

**Comment by ohmeow** [comment]
This looks identical to  #1292 .  Feel free to merge my comments in with that one.

**Comment by sgugger** [comment]
Solved by #1293
